django_project/certification/views/certificate.py

```diff
@@ -125,7 +125,8 @@ def get_object(self, queryset=None):
         if queryset is None:
             queryset = self.get_queryset()
             certificateID = self.kwargs.get('id', None)
-            if certificateID:
+            certificate = Certificate.objects.all().values_list('certificateID', flat=True)
+            if certificateID in certificate:
```

Comment:
why not rather use the `get or 404` pattern here? I think we should rather use the DB to search for a record than return all the objects and search in python...